src/Core/Lock/ILockDriver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Friendica\Core\Lock;
+
+/**
+ * Lock Driver Interface
+ *
+ * @author Philipp Holzer <admin@philipp.info>
+ */
+interface ILockDriver
+{
+	/**
+	 * Checks, if a key is currently locked to a or my process
+	 *
+	 * @param string $key 		The name of the lock
+	 * @return bool
+	 */
+	public function isLocked($key);
+
+	/**
+	 *
+	 * Acquires a lock for a given name
+	 *
+	 * @param string  $key      The Name of the lock
+	 * @param integer $timeout  Seconds until we give up
+	 *
+	 * @return boolean Was the lock successful?
+	 */
+	public function acquire($key, $timeout = 120);
```

Comment:
I'm sorry, I got confused about the method naming, here's what I'd like to see:
- `Lock::acquire()`
- `Lock::release()`
- `LockDriver::acquireLock()`
- `LockDriver::releaseLock()`

Please tell me if it makes as much sense to you as it does to me.

Reply:
I accidentally changed the wrong methods.
So yes, it makes sense for me too :-)